torch/_dynamo/config.py

```diff
@@ -516,12 +513,11 @@ def default_debug_dir_root():
     "storage_offset",
     "layout",
     "data",
+    "is_neg",
+    "is_conj",
+    "is_pinned",
```

Comment:
shouldn't we also ban is_contiguous?

Reply:
Actually I'm confused. In the current version of the PR, why does the backward speculation fail on the is_contiguous call?

Reply:
Yeah I think this PR is wrong. We should still ban is_contiguous here.

Reply:
Oh, the current logic does not ban `is_contiguous`; instead, we assume `grad_output` is valid and speculate the backward pass from there. If that approach fails, we catch the exception and retry with a rewritten backward pass with inserting `grad_output.contiguous()`.

From my understanding, your idea is to explicitly ban `is_contiguous`, catch the strict mode exception when it’s called, and then speculate the rewritten backward pass. That approach seems safer, and I agree with it.